siteapp/tests.py

```diff
@@ -266,72 +285,75 @@ def setUp(self):
         )
         load_modules().handle() # load system modules
 
-        AppSource.objects.create(
+        AppSource.objects.get_or_create(
             slug="project",
             spec={ # contains a test project
                 "type": "local",
                 "path": "fixtures/modules/other",
             },
             trust_assets=True
-        )\
-            .add_app_to_catalog("simple_project")
+        )[0].add_app_to_catalog("simple_project")
 
         # Create a default user that is a member of the organization.
         # Log the user into the test client, which is used for API
         # tests. The Selenium tests require a separate log in via the
         # headless browser.
 
         # self.user = User.objects.create_superuser(
-        self.user = User.objects.create(
+        self.user = wait_for_sleep_after(lambda: User.objects.get_or_create(
             username="me",
             email="test+user@q.govready.com",
             is_staff=True
-        )
+        )[0])
         self.user.clear_password = get_random_string(16)
         self.user.set_password(self.user.clear_password)
         self.user.user_permissions.add(Permission.objects.get(codename='view_appsource'))
-        self.user.save()
+        wait_for_sleep_after(lambda: self.user.save())
         self.user.reset_api_keys()
         self.user.user_permissions.add(Permission.objects.get(codename='view_appsource'))
         self.client.login(username=self.user.username, password=self.user.clear_password)
 
         # Create a Portfolio and Grant Access
-        portfolio = Portfolio.objects.create(title=self.user.username)
+        portfolio = Portfolio.objects.get_or_create(title=self.user.username)[0]
```

Comment:
I'm curious, what's the need for the `[0]` at the end? Shouldn't `get_or_create` only return a single object? Same question throughout this file.

Reply:
get_or_create  returns a tuple with the 0 index being the actual portfolio in this case